### PR TITLE
Add benchmark for DynamoDB serialization/deserialization

### PR DESCRIPTION
**Pull request by jdisanti**
More work towards #161. This adds a benchmark for JSON serialization/deserialization for real-world DynamoDB calls.

Sample output:
```
$ cargo bench
ser_deser_bench         time:   [224.39 us 226.10 us 228.14 us]
Found 5 outliers among 100 measurements (5.00%)
  3 (3.00%) high mild
  2 (2.00%) high severe
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rcoh — APPROVED** [review]
not sure if intentionally or unintentionally, you're actually testing a lot more than just the serializers. If you just want to benchmark the serializer, I would write something much closer to what the protocol tests generate and use `make_operation` and `ParseResponse` directly. (Note that in the case of dynamo, you can use `ParseStrictResponse` since none of the responses are streaming)

**Comment by jdisanti** [comment]
Definitely was unintentionally testing more than I wanted to, but I wasn't aware of how to do the parsing directly. Thanks for the pointers! I updated the PR with a much nicer implementation that should give us better information.

**Review by rcoh — COMMENTED** [review]
Looks good, removing async will also simplify things.

We should also add the benches folder to the copy step in build.gradle so that CI will at least compile these

File: aws/sdk/build.gradle.kts [hunk 1]
```diff
@@ -240,7 +245,7 @@ tasks.register<Exec>("cargoCheck") {
     workingDir(sdkOutputDir)
     // disallow warnings
     environment("RUSTFLAGS", "-D warnings")
-    commandLine("cargo", "check")
+    commandLine("cargo", "check", "--lib", "--tests", "--benches")
```
Comment (rcoh) [inline aws/sdk/build.gradle.kts:248]: Since CI doesn't actually hit this, we should also update the CI job

**Review by rcoh — APPROVED** [review]
LGTM!

**Pull request merged**